ast/ddl.go

```diff
@@ -1209,6 +1209,56 @@ func (n *CreateViewStmt) Accept(v Visitor) (Node, bool) {
 	return v.Leave(n)
 }
 
+// IndexLockAndAlgorithm stores the algorithm option and the lock option.
+type IndexLockAndAlgorithm struct {
+	node
+
+	LockTp      *LockType
+	AlgorithmTp *AlgorithmType
```

Comment:
I suggest storing these as values instead of pointers.

```go
	LockTp      LockType
	AlgorithmTp AlgorithmType
```

Treat `ALGORITHM = DEFAULT` and `LOCK = DEFAULT` as unspecified.